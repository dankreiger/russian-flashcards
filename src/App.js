import React, { Component } from 'react';
import Flashcard from './Flashcard';
import vocabulary from './vocabulary';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = { flipped: false, card: 0 };
  }

  flipCard = () => {
    this.setState({flipped: !this.state.flipped});
  }

  nextCard = () => {
    this.setState({flipped: false, card: this.state.card === vocabulary.length - 1 ? 0 : this.state.card + 1});
  }

  render() {
    const { card, flipped } = this.state;

    return (
      <div className="App">
        <Flashcard card={card} flipped={flipped} />
        <div className="flex-buttons">
          <button className="btn blue" onClick={this.flipCard}>
            <span>Flip</span>
          </button>
          <button className="btn" onClick={this.nextCard}>
            <span>Next</span>
          </button>
        </div>
      </div>
    );
  }
}

export default App;
