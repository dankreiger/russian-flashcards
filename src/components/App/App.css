.App {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: space-around;
      -ms-flex-pack: space-around;
          justify-content: space-around;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.starting-language-menu {
  -ms-flex-item-align: flex-end;
    align-self: flex-end;
}

select {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;  
}
