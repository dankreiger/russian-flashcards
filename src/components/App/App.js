import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import Flashcard from '../Flashcard/Flashcard';
import Buttons from '../Buttons/Buttons';
import Artyom from 'artyom.js';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { flipped: false, card: 0, currentLang: "English" };

    this.vocabulary = this.props.vocabulary;
  }

  flipCard = () => {
    this.setState({flipped: !this.state.flipped});
  }

  nextCard = () => {
    this.setState({flipped: this.state.currentLang === "Russian", card: this.state.card === this.vocabulary.length - 1 ? 0 : this.state.card + 1});
  }

  sayWord = () => {
    let artyom = new Artyom();
    artyom.say(this.vocabulary[this.state.card].word, {
      lang:"ru-RU"
    });
  }

  render() {
    const { card, flipped } = this.state;

    return (
      <div className="App">
        <div className="links-container">
          <Link to="/">Back</Link>
          <div className="starting-language-menu">
            <label>Front Side </label>
            <select onChange={e => this.setState({currentLang: e.target.value, flipped: e.target.value === "Russian"})}>
              <option>English</option>
              <option>Russian</option>
            </select>
          </div>
        </div>
        <Flashcard card={this.vocabulary[card]} flipCard={this.flipCard} flipped={flipped} />
        <Buttons card={this.vocabulary[card]} sayWord={this.sayWord} flipCard={this.flipCard} nextCard={this.nextCard}/>
      </div>
    );
  }
}

export default App;
