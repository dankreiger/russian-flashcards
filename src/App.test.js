import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


import App from './App';

describe('<App />', () => {
  let app;
  beforeEach(() => {
    app = shallow(<App />);
  });

  it('renders correctly', () => {
    expect(toJson(app)).toMatchSnapshot();
  });

  it('renders a <Flashcard /> component', () => {
    expect(app.find('Flashcard').exists()).toBe(true);
  });
});
