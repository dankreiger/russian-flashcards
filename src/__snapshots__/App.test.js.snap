// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<App>
  <div
    className="App"
  >
    <Flashcard
      card={0}
      flipCard={[Function]}
      flipped={false}
    >
      <div
        className="Flashcard"
        onClick={[Function]}
      >
        <Front
          card={
            Object {
              "phonetic": "[dá]",
              "translation": "Yes",
              "word": "Да",
            }
          }
          flipped={false}
        >
          <div
            className="front"
          >
            <p
              className="lead"
            >
              Да
            </p>
            <p
              className="phonetic"
            >
              [dá]
            </p>
          </div>
        </Front>
        <Back
          card={
            Object {
              "phonetic": "[dá]",
              "translation": "Yes",
              "word": "Да",
            }
          }
          flipped={false}
        >
          <div
            className="back hidden"
          >
            <p
              className="lead"
            >
              Yes
            </p>
            <p
              className="phonetic"
            >
              [dá]
            </p>
          </div>
        </Back>
      </div>
    </Flashcard>
    <div
      className="flex-buttons"
    >
      <button
        className="btn blue flipCard"
        onClick={[Function]}
      >
        <span>
          Flip
        </span>
      </button>
      <button
        className="btn nextCard"
        onClick={[Function]}
      >
        <span>
          Next
        </span>
      </button>
    </div>
  </div>
</App>
`;
